se/msg.py

```diff
@@ -114,8 +114,11 @@ def readMsg(inFile, recFile, mode):
             else:
                 msg += nextByte
         msg = msg[:-magicLen]  # strip the magic number from the end
-    if len(msg) > 0:  # don't log zero length messages
-        logger.message("-->", dataInSeq, magic + msg, inFile.name)
+
+    if len(msg) < 1:  # don't log zero length messages
```

Comment:
good catch :)

However, don't you just want to just change the condition?  By adding the return you are changing what data is recorded.  Is this desirable?  cc @jbuehl 

That said, a more pythonic test would be something like 
```
if msg:
    logger.message("-->", dataInSeq, magic + msg, inFile.name)
```
https://www.python.org/dev/peps/pep-0008/#programming-recommendations

Reply:
actually, by not adding the return we are already changing the data recorded, as invalid messages get now get added to the log as an empty message.
e.g. if the message contains only half a message (e.g. a pcap file that starts midway tranmission). Our current code discards this message as the CRC does not match. We now have a zero length message which does get added to the rec log. This is not present in the json and technically was not present in the source.

As for the python style; the original code was explicitly checking for the length, so I stuck with that. If @jbuehl can agree that the existence of a message is good enough; I'll change it as part of this PR.

Reply:
right now, for me, i use it to log all data that goes through semonitor, as the output json is not always accurate. For example. Message XYZ is currently unknown, so gets discarded or garbled in the log (timestamps in the event log for example), but does get logged in the rec file. If we now properly decode these logs however, I cannot go and backtrack my json logs. With the rec logs however I can do exactly this.

Reply:
as for the zero length message, right now, (and this is from 2 month old memories) if i have a rec file and i run it through semonitor again (which, with the rec option should produce the exact same output) it currently does not, but instead adds a new zero length message to the start. This zero length message was not there before, but it is now and thus the rec files (in and out) do no longer match.

I think if you run the new test case attached to this issue, with the old data, you'll see what i mean.

Again, this happens due to half message starts. With the replay (rec in -> rec out; cmp) i have not noticed any difference after this change however.

Reply:
Assuming you are not using an active mode @oliv3r and you want an exact copy of the input file, why not put `tee` in the pipe before semonitor.py to send it to a file?  The record file would still be useful for recording both sides of the conversation in active mode.

Reply:
sadly, your assumptions are invalid :p i'm using master/active rs-485 mode :)

but i'm thinking of cooking up a simple daemon that does the tcp -> rs-485 step (and decodes the packet (not the payload) allowing for multiple users (semonitor, other tools) to talk to the inverter simultaneously.